### PR TITLE
SI-6099 - Scaladoc for scala.concurrent incomplete

### PR DESCRIPTION
**Pull request by phaller**
Complete and improve doc comments of `scala.concurrent`:
- Skip generating docs for `scala.concurrent.impl` package
- Added doc comment to `Awaitable` trait
- Rewrote doc comments of `Awaitable.ready`/`Await.ready` and `Awaitable.result`/`Await.result`
- Reworded doc comment about managed blocking in the `Await` object
- Fix doc comment of `scala.concurrent.blocking`

Review by @heathermiller and @axel22

### CODE REVIEW

File: src/library/scala/concurrent/package.scala [hunk 1]
```diff
@@ -87,19 +87,21 @@ package concurrent {
...
-     *        [[Duration.Inf]] for unbounded waiting, or a finite positive
-     *        duration
-     * @return the value if the Awaitable was successful within the specific maximum wait time
-     * @throws InterruptedException     if the wait call was interrupted
-     * @throws TimeoutException         if after waiting for the specified time this Awaitable is still not ready
-     * @throws IllegalArgumentException if `atMost` is [[Duration.Undefined]]
+     * Await and return the result (of type `T`) of an `Awaitable`.
+     * 
+     * Although this method is blocking, the internal use of [[scala.concurrent.blocking blocking]] ensures that
+     * the underlying [[ExecutionContext]] is prepared to properly manage the blocking.
```
Comment (heathermiller) [inline src/library/scala/concurrent/package.scala:93]: Nitpick-- with respect to: "is prepared to properly manage the blocking" This is still a bit vague, no?

**Comment by jsuereth** [comment]
This one needs a rebase against 2.10.0-wip

File: src/library/scala/concurrent/package.scala (hunk 1 above)
Comment (axel22) [inline src/library/scala/concurrent/package.scala:93]: Maybe "to properly detect blocking and ensure that there are no deadlocks."?

**Comment by axel22** [comment]
LGTM

File: src/library/scala/concurrent/package.scala (hunk 1 above)
Comment (heathermiller) [inline src/library/scala/concurrent/package.scala:93]: Yep, better :)

**Comment by heathermiller** [comment]
Ok- I can rebase this and add Alex's rewording now, Philipp is currently in a training. I'll submit a separate pull.

**Comment by heathermiller** [comment]
All is corrected in pull https://github.com/scala/scala/pull/1486

**Comment by heathermiller** [comment]
Btw, LGTM :)

**Pull request closed without merging**